sources/packages/backend/libs/services/src/queue/queue.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { QueueConfiguration } from "@sims/sims-db";
+import { Repository } from "typeorm";
+import { QueueModel } from "./model/queue.model";
+
+@Injectable()
+export class QueueService {
+  constructor(
+    @InjectRepository(QueueConfiguration)
+    private queueConfigurationRepo: Repository<QueueConfiguration>,
+  ) {}
+
+  /**
+   * Get all queue name and configurations.
+   * @returns queue configuration.
+   */
+  async getAllQueueConfigurations(): Promise<QueueConfiguration[]> {
+    return this.queueConfigurationRepo.find({
+      select: {
+        queueName: true,
+        queueConfiguration: {
```

Comment:
Typeorm does not actually support this type of query, as far as I know.
Even not the best, I would prefer having the JSONB column mapped as unknown and forcing the consumer to do the cast to `QueueConfigurationDetails` when needed.